### ISSUE TITLE
top_k_acc is higher, but sglang accept len only has 1.26

### ISSUE THREAD

**Issue by Dawson-Ren** [NickL77/BaldEagle#17]
![Image](https://github.com/user-attachments/assets/7cee4e59-70c7-4c9a-9efd-eceb3794d57d)

![Image](https://github.com/user-attachments/assets/2c88a021-0f51-42df-8218-2ba2aa58fb4c)

![Image](https://github.com/user-attachments/assets/ee2af3d3-85aa-4536-a07b-aa638bfd5bb4)

**Comment by Dawson-Ren** [comment]
> I'll need more details on what you're trying to train to help out. It does seem that your data size is quite small (5 epochs only ~12k steps). With top_k accuracy being that high, I'd expect toe be getting at least accept_len > 2. You might be overfitting to a small, specific dataset.

There are 40k training samples, but my test data is also sampled from the training data, so it should also achieve a pretty good acceleration effect.

Also, I retrained the model with 100k data, and the top-k accuracy quickly reached 90%, but the sglang accept length is 1.2.

![Image](https://github.com/user-attachments/assets/ead18c1b-f59e-43f2-86b5-3631341c0a15)
training_args = TrainingArguments(
    output_dir=f"./hf_trainer_output_dir/{swanlab_name}/",
    num_train_epochs=5,
    gradient_accumulation_steps=16,
    per_device_train_batch_size=1,
    per_device_eval_batch_size=1,
    remove_unused_columns=False,
    bf16=True,
    fp16=False,
    dataloader_num_workers=4,
    warmup_ratio=0.01,
    learning_rate=1e-4,  # 1e-3
    lr_scheduler_type="constant",  # Placeholder, we override it in the trainer
    max_grad_norm=0.5,  # 1
    adam_beta1=0.9,  # 0.9
    adam_beta2=0.95,  # 0.999
    weight_decay=1e-2,
    eval_strategy="steps",
    logging_steps=32,
    eval_steps=64,
    save_strategy="steps",
    save_steps=0.1,  # saves every 10% of training
    save_total_limit=3,
    report_to="swanlab",
    run_name=f"{swanlab_name}_llama_eagle_train",
)

trainer = EagleTrainer(
    model=draft_model,
    head=head,
    args=training_args,
    train_dataset=eagle_train_dataset,
    eval_dataset=eagle_test_dataset,
    data_collator=eagle_collator,
    min_lr_ratio=0.5,  # Custmer lr scheduler param
)

trainer.train()

**Comment by Greatpanc** [comment]
I also encountered this problem. My top_1_acc reached 0.97, but the actual acceptance length was about 1.56
